### ISSUE TITLE
com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Read access is allowed from inside read-action only

### ISSUE THREAD

**Issue by piotr-dobrogost** [vlascik/Tabdir#5]
Thank you very much for maintaining and extending Tabdir plugin!

I'm using it in PyCharm and after clicking any editor tab the IDE reports the following exception:

```
com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Read access is allowed from inside read-action only (see Application.runReadAction()); see https://jb.gg/ij-platform-threading for details
Current thread: Thread[#117,DefaultDispatcher-worker-23,5,main] 1668521142 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[#50,AWT-EventQueue-0,6,InnocuousThreadGroup] 1488975051
	at com.intellij.util.concurrency.ThreadingAssertions.createThreadAccessException(ThreadingAssertions.java:177)
	at com.intellij.util.concurrency.ThreadingAssertions.softAssertReadAccess(ThreadingAssertions.java:129)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:915)
	at com.intellij.util.indexing.FileBasedIndexScanUtil.ensureUpToDate(FileBasedIndexScanUtil.java:48)
	at com.intellij.util.indexing.FileBasedIndexScanUtil.processFilesContainingAnyKey(FileBasedIndexScanUtil.java:241)
	at com.intellij.util.indexing.FileBasedIndexImpl.processFilesContainingAnyKey(FileBasedIndexImpl.java:1175)
	at com.intellij.psi.search.FilenameIndex.getVirtualFilesByNames(FilenameIndex.java:233)
	at com.intellij.psi.search.FilenameIndex.getVirtualFilesByName(FilenameIndex.java:88)
	at ru.crazycoder.plugins.tabdir.SameFilenameTitleProvider.titleWithDiffs(SameFilenameTitleProvider.java:124)
	at ru.crazycoder.plugins.tabdir.SameFilenameTitleProvider.getEditorTabTitleInternal(SameFilenameTitleProvider.java:69)
	at ru.crazycoder.plugins.tabdir.SameFilenameTitleProvider.getEditorTabTitle(SameFilenameTitleProvider.java:55)
	at com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider$getEditorTabTitleAsync$2.invoke(EditorTabTitleProvider.kt:29)
	at com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider$getEditorTabTitleAsync$2.invoke(EditorTabTitleProvider.kt:28)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:339)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invokeSuspend(coroutines.kt:232)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:261)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:231)
	at com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider.getEditorTabTitleAsync$suspendImpl(EditorTabTitleProvider.kt:28)
	at com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider.getEditorTabTitleAsync(EditorTabTitleProvider.kt)
	at com.intellij.openapi.fileEditor.impl.EditorTabPresentationUtil.getCustomEditorTabTitleAsync(EditorTabPresentationUtil.kt:54)
	at com.intellij.openapi.fileEditor.impl.EditorTabbedContainer$insertTab$3.invokeSuspend(EditorTabbedContainer.kt:240)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
```

```
PyCharm 2024.2.1 (Professional Edition)
Build #PY-242.21829.153, built on August 29, 2024
Runtime version: 21.0.3+13-b509.11 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.wl.WLToolkit
Linux 6.10.7-200.fc40.x86_64
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 4
Registry:
  debugger.watches.in.variables=false
  ide.experimental.ui=true
  i18n.locale=
Non-Bundled Plugins:
  com.jetbrains.jbre.bintray.downloader (2.2)
  A prevent pinned tabs from closing plugin - by momomo.com (17.850)
  com.intellij.ideolog (242.21829.162)
  CMD Support (1.0.5)
  Key Promoter X (2024.2.0)
  org.sylfra.idea.plugins.linessorter (1.0.1)
  vlascik.plugins.tabdir-redux (2.0.3)
  GrepConsole (13.1.0-IJ2023.3)
Current Desktop: KDE

```

**Comment by Mesqalito** [comment]
Encountered same problem in WebStorm 2024.3. Any updates?

**Comment by vlascik** [comment]
Should be fixed in v2.0.4, but then again, I have pretty much no idea what I'm doing here 😄.

New version is in `Releases` and uploaded to Jetbrains, it should be available from inside the IDE either today or tomorrow. If it doesn't fix your problems, please reopen this issue.
